integration-tests/suites/common/executor.go

```diff
@@ -150,6 +159,11 @@ func (e *executor) Exec(args ...string) (string, error) {
 func (e *executor) ExecRetry(args ...string) (res string, err error) {
 	maxAttempts := 3
 	attempt := 0
+
+	if args[0] == RuntimeCommand && RuntimeAsRoot {
+		args = append([]string{"sudo"}, args...)
+	}
```

Comment:
Could we move this logic inside the `RunCommand` function? Think it would make sense to have it as close to actually running the command as possible. It's also duplicated in the `Exec` function, which is not ideal IMO.

Reply:
Unfortunately we can't easily put it there, because by that point the command is fully constructed and is prepended with various other parts of the command to run on the right VM (gcloud, ssh, etc) so we need to prepend sudo to the start of the command that will be executed on the remote host and unfortunately that does come with a little duplication currently

Reply:
That's unfortunate but understandable.